packages/flutter_android_lifecycle/android/src/main/java/io/flutter/embedding/engine/plugins/lifecycle/FlutterLifecycleAdapter.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.plugins.lifecycle;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.Lifecycle;
+
+public class FlutterLifecycleAdapter {
+  @NonNull private final Lifecycle lifecycle;
+
+  public FlutterLifecycleAdapter(@NonNull Object reference) {
+    if (!(reference instanceof HiddenLifecycleReference)) {
```

Comment:
It's possible that this is going to be compiled with a Flutter Engine version that does not have the HiddenLifecycleReference (e.g if a migrated plugin is used with today's stable).

I think we should use reflection to access this type? (can we make sure proguard doesn't rename it?)